### PR TITLE
🌱 Update CI VSphere template to k8s 1.28.0

### PR DESCRIPTION
**Pull request by zhanggbj**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of #2203

### CODE REVIEW

**Comment by chrischdi** [comment]
Should we also add the new images to the `README.md` in this PR?

@sbueringer : could you upload them to the CI infrastructure:

https://storage.googleapis.com/capv-templates/v1.28.0/flatcar-stable-3510.2.6-kube-v1.28.0.ova

https://storage.googleapis.com/capv-templates/v1.28.0/ubuntu-2004-kube-v1.28.0.ova

**Comment by chrischdi** [comment]
Also:

photon-3, ubuntu-1804 and ubuntu-2204 are missing at the uploaded OVA's :-)

**Comment by zhanggbj** [comment]
@chrischdi yes, I'm still building photon-3, ubuntu-1804 and ubuntu-2204, will upload them later. If CI test is not so urgent, I can also update the `README.md` in this PR, otherwise, we can do it in a separate PR

**Comment by sbueringer** [comment]
> @chrischdi yes, I'm still building photon-3, ubuntu-1804 and ubuntu-2204, will upload them once done. If CI test is not so urgent, I can also update the `README.md` in this PR laster :) otherwise, we can do it in a separate PR

I think it's probably fine to do it all in this PR

**Comment by sbueringer** [comment]
/test pull-cluster-api-provider-vsphere-e2e-main

Just to get more data, based on logs it already failed

File: test/e2e/config/vsphere-ci.yaml [hunk 1]
```diff
@@ -131,8 +131,8 @@ providers:
           - sourcePath: "../data/shared/v1beta1_provider/metadata.yaml"
 
 variables:
-  KUBERNETES_VERSION: "v1.27.3"
-  CPI_IMAGE_K8S_VERSION: "v1.27.0"
+  KUBERNETES_VERSION: "v1.28.0"
+  CPI_IMAGE_K8S_VERSION: "v1.28.0"
```
[the remark this answers is not in the document]
Comment (sbueringer) [reply test/e2e/config/vsphere-ci.yaml:135]: Ah good catch. Yeah let's go with v1.27.0 for this PR and once the CCM release is out we create a follow-up PR

File: test/e2e/config/vsphere-dev.yaml [hunk 2]
```diff
@@ -134,8 +134,8 @@ providers:
           - sourcePath: "../data/shared/v1beta1_provider/metadata.yaml"
 
 variables:
-  KUBERNETES_VERSION: "v1.27.3"
-  CPI_IMAGE_K8S_VERSION: "v1.27.0"
+  KUBERNETES_VERSION: "v1.28.0"
+  CPI_IMAGE_K8S_VERSION: "v1.28.0"
```
[the remark this answers is not in the document]
Comment (zhanggbj) [reply test/e2e/config/vsphere-dev.yaml]: @chrischdi Thanks for catching it!

**Comment by chrischdi** [comment]
Tests are green :-) 🎉 

/lgtm
/approve

**Pull request merged**